src/js/actions/export.js

```diff
@@ -483,38 +521,42 @@ define(function (require, exports) {
                         log.debug("Export: Starting Generator...");
                         return _setGeneratorStatus(true);
                     }
+                    return false;
                 })
-                .then(function () {
-                    _exportService = new ExportService();
-                    return _initService.call(this)
-                        .catch(function (e) {
-                            throw new Error("ExportService.init explicitly returned: " + e.message);
-                        });
+                .then(function (generatorJustStarted) {
+                    if (generatorJustStarted) {
+                        // ONLY IF we are having to force enable generator, which shouldn't be often
+                        // The new port isn't configured immediately, and start up is not synchronous
+                        // This is partially caused by the race in generator-core startWebsocketServer
+                        return Promise.delay(3000).then(_getWebSocketPort);
```

Comment:
Additionally, with this PR, we do a single retry of the whole she-bang, so if the plugin is NOT registered in the first 3 after starting generator, it could possibly get resolved on the second try.